core_info.c

```diff
@@ -84,6 +84,8 @@ static core_info_cache_list_t *core_info_cache_list_new(void);
 static void core_info_cache_add(core_info_cache_list_t *list, core_info_t *info,
       bool transfer);
 
+static core_info_state_t core_info_st;
```

Comment:
This PR looks fine to me, but I'd be somewhat inclined to initialise `core_info_st` explicitly (just for any oddball compilers that don't follow standards and omit the automatic zero initialisation) - i.e.:

```c
static core_info_state_t core_info_st = {
#ifdef HAVE_COMPRESSION
   NULL,
#endif
   NULL,
   NULL,
   NULL
}
```

It shouldn't be necessary on sane platforms, though...